server/s3_common_utilities.cc

```diff
@@ -30,6 +30,29 @@
 
 namespace S3CommonUtilities {
 
+// Special char hash table
```

Comment:
add a comment and list all special chars handled here
// '!' -> special_char[30]
its better to make it specific to canonical id name it canonical_special_char as we are not required to handle all spcial chars like '/'

Reply:
done